tapscript/tx.go

```diff
@@ -71,65 +71,99 @@ func virtualTxInPrevOut(root mssmt.Node) *wire.OutPoint {
 	)
 }
 
+// virtualGenesisTxIn computes the single input of a Taproot Asset virtual
+// transaction where the asset is both a genesis asset and part of an asset
+// group. The input MS-SMT commits to the genesis asset with witnesses removed.
+func virtualGenesisTxIn(newAsset *asset.Asset) (*wire.TxIn, mssmt.Tree, error) {
+	inputTree := mssmt.NewCompactedTree(mssmt.NewDefaultStore())
+
+	// TODO(bhandras): thread the context through.
+	ctx := context.TODO()
+
+	// For genesis assets with a group key, we must clear the witness field
+	// before building the input MS-SMT. We use this copy of the genesis
+	// asset as the input asset for the virtual transaction.
+	copyWithoutWitness := newAsset.Copy()
+	for i := range copyWithoutWitness.PrevWitnesses {
+		copyWithoutWitness.PrevWitnesses[i].TxWitness = nil
+	}
+
+	key := asset.ZeroPrevID.Hash()
```

Comment:
Yeah, this looks correct I think! During VM execution the prev witness will be zero and point back to this zero-id.